catalyst/loggers/mlflow.py

```diff
@@ -0,0 +1,243 @@
+
+
+def _mlflow_log_dict(
+        dictionary: Dict[str, Any], prefix: str = "", log_type: Optional[str] = None
+):
+    """The function of MLflow. Logs any value by its type from dictionary recursively.
+
+    Args:
+        dictionary: Values to log as dictionary.
+        prefix: Prefix for parameter name (if the parameter is composite).
+        log_type: The entity of logging (param, metric, artifact, image, etc.).
+
+    Raises:
+        ValueError: If meets unknown type or log_type for logging in MLflow
+            (add new case if needed).
+    """
+    for name, value in dictionary.items():
+        if name in EXCLUDE_PARAMS:
+            continue
+
+        name = name.replace("*", "")
+        if prefix not in STAGE_PARAMS and prefix:
+            name = f"{prefix}/{name}"
+
+        if log_type == "dict":
+            mlflow.log_dict(dictionary, name)
+        elif isinstance(value, dict):
+            _mlflow_log_dict(value, name, log_type)
+        elif log_type == "param":
+            try:
+                mlflow.log_param(name, value)
+            except mlflow.exceptions.MlflowException:
+                continue
+        elif isinstance(value, (Directory, File)) or log_type == "artifact":
+            mlflow.log_artifact(value)
+        elif isinstance(value, Number):
+            mlflow.log_metric(name, value)
+        else:
+            raise ValueError(f"Unknown type of logging value: {type(value)}")
+
+
+class MLflowLogger(ILogger):
+    """MLflow logger for parameters, metrics, images and other artifacts.
+
+    MLflow documentation: https://mlflow.org/docs/latest/index.html.
+
+    Notebook API example:
+
+    .. code-block:: python
+
+        from catalyst import dl
+
+        class CustomSupervisedRunner(dl.IRunner):
+            def get_engine(self) -> dl.IEngine:
+                return dl.DeviceEngine("cpu")
+
+            def get_loggers(self):
+                return {
+                    "console": dl.ConsoleLogger(),
+                    "mlflow": dl.MLflowLogger(experiment="test_exp", run="test_run")
+                }
+
+        runner = CustomSupervisedRunner().run()
+        model = runner.model
+
+    Config API example:
+
+    .. code-block:: yaml
+
+        loggers:
+            mlflow:
+                _target_: MLflowLogger
+                experiment: test_exp
+                run: test_run
+        ...
+    """
+
+    def __init__(
+        self,
+        experiment: str,
+        run: Optional[str] = None,
+        tracking_uri: Optional[str] = None,
+        registry_uri: Optional[str] = None,
+    ) -> None:
+        """
+        Args:
+            experiment: Name of the experiment in MLflow to log to.
+            run: Name of the run in MLflow to log to.
+            tracking_uri: URI of tracking server against which
+                to log run information related.
+            registry_uri: Address of local or remote model registry server.
+        """
```

Comment:
could you please move it to the class description?
example: https://github.com/catalyst-team/catalyst/blob/dev/catalyst/metrics/_accuracy.py#L12#L34

Reply:
done